### PR TITLE
add an argument iso_week to incidence and update documents

### PR DESCRIPTION
**Pull request by caijun**
@thibautjombart I have added the argument `iso_week` to `incidence.Date()` and `incidence.POSIXt()`, and updated the documents, DESCRIPTION (including version) and NEWS.md. 

I have locally test the package, test plotting failed. Since the previous implementation is kept, the previous rds files for test-plot.R should be restored. After you have restored the rds files, I think the test plotting units will pass and you can publish the new version of incidence package on CRAN.

### CODE REVIEW

**Comment by thibautjombart** [comment]
Thanks for this. I'll restore the previous plots and check if the branch passes the checks.

File: R/incidence.R [hunk 1]
```diff
@@ -140,8 +146,10 @@ incidence.integer <- function(dates, interval = 1L, groups = NULL,
     first.date <- min(dates)
     last.date <- max(dates)
     interval <- as.integer(round(interval))
-    if (interval == 7L) {
-      first.date <- 0L
+    if ("iso_week" %in% names(dots)) {
+      if (interval == 7L & dots$iso_week == TRUE) {
```
Comment (caijun) [inline R/incidence.R]: yes, && is appropriate for programming control-flow and typically preferred in if clauses. I will fix it.

**Comment by thibautjombart** [comment]
Great, previous plots restored and now tests passing again. Gave it a few tries, seems to work like a charm :) 

One remaining thing I'm not sure about. Regarding the labels. Because `as.data.frame` returns the isoweeks I think it should be possible to plot dates as isoweeks too. But I'm not sure if `ggplot2` handles them? If so maybe yet another argument like `iso_week_labels = iso_week` by default?

**Comment by caijun** [comment]
yes, I also notice this feature for plotting incidence. `iso_week_labels = iso_week` can control the x-axis label showing ISOweek character instead of Date character.

**Comment by thibautjombart** [comment]
Cool. I think to make it consistent the `$dates` should be changed as well to reflect the  ISO. So basically, the default for weekly incidence will be ISO, labelled as such. Then the user will have the option of using ISO while keeping dates, or not using ISO at all. Makes sense?

**Comment by caijun** [comment]
So you mean when computing ISOweek-based weekly incidence, only `isoweeks` will be returned? I prefer to return both `dates` and `isoweeks`.  As described in the function help document, in such a case, the `dates` will be the first days of corresponding isoweeks, which is necessary based on my data analysis experience. However, adding an argument  `iso_week_labels = iso_week` to `plot.incidence()` can help the user switch between two kinds of x-axis labels (Date and ISOweek) easily.

**Comment by caijun** [comment]
Besides, in practical plot, the user may also want to plot x-axis label with dates (such as isoweek starting day or isoweek ending day) for ISO week-based incidence data. I have seen such plots while reading papers.

**Comment by thibautjombart** [comment]
Yes, the essential bit is the graphics. In the case where ISOweeks is used, the plot should reflect this and label the x-axis accordingly. Leaving an option to use regular dates otherwise.

Unrelated, but it seems the coverage stats are now reflecting this current branch, which it should not, but it reminds me to ask: we need unit tests covering the new lines of code ;) Think you can PR these alongside the new plot argument?

**Comment by caijun** [comment]
I have thought your suggestion throughly and think that providing an option for labeling ISOweeks on the x-axis accordingly is a better choice. I will implement it quickly. The unit tests will also be provided.

**Comment by thibautjombart** [comment]
Awesome!

**Pull request merged**